### DISCUSSION TITLE
Allow login with service tokens

### DISCUSSION THREAD

**Discussion by dankolbman** [nih-cfde/published-documentation#188] [Feature Requests]
## Request
Currently, it would appear that login only works via logging the user into globus and capturing the returned token. This prevents further automation of the submission process as we will always need a user with the appropriate credentials to be present to log in through the browser session.

## Implementation

There are a number of ways to address this. The most common approach for those already using OAuth2 type flows is to use the `client_credentials` flow which allows a service to be registered with the system and receive a key pair in return. This keypair is then exchanged with an OAuth2 endpoint to provide a bearer token which may then be used with the system in a similar manner as the implicit grant tokens the user would get in the browser.

## Background
The Kids First DCC runs a release process to publish new data to its users. This process is highly automated to ensure that we may push out new data to our users quickly and frequently. There are two phases to the process:
1) staging the release
2) publishing the release

An internal member of the DCC will begin the release by selecting a number of Studies to be published. From here, all computations will done to 'stage' the release for our downstream systems. The initial CFDE submission would occur here, if we could run `cfde-submit` automatically by our release system.
Once the computations are complete, the release will be said to be 'staged'. At this point, the DCC will review the release before continuing (this coincides nicely with the CFDE review process).
Once checks of the staged release are complete, the DCC initiates an automated procedure to publish the release. This procedure makes the staged data available and accessible by users on our downstream system. This process would also ideally include approving the CFDE submission.

**Comment by ACharbonneau** [comment]
Thank you! I'll bring this up with the development team
